rpc/server.go

```diff
@@ -39,178 +40,84 @@ func NewRpcServer(nitroClient *nitro.Client, chainId *big.Int, logger zerolog.Lo
 
 	opts := &server.Options{}
 	ns, err := server.NewServer(opts)
-	handleError(err)
+	if err != nil {
+		panic(err)
+	}
 	ns.Start()
 
 	nc, err := nats.Connect(ns.ClientURL())
-	handleError(err)
+	if err != nil {
+		panic(err)
+	}
 
 	rs := &RpcServer{natstrans.NewNatsConnection(nc), ns, nitroClient, chainId, logger}
-	rs.registerHandlers()
+	err = rs.registerHandlers()
+	if err != nil {
+		panic(err)
+	}
+
 	return rs
 }
 
-// registerHandlers registers the handlers for the rpc server
-func (rs *RpcServer) registerHandlers() {
-	err := rs.connection.Subscribe(serde.DirectFundRequestMethod, func(data []byte) []byte {
-
+func subcribeToRequest[T serde.RequestPayload, U serde.ResponsePayload](rs *RpcServer, method serde.RequestMethod, processPayload func(T) U) error {
+	return rs.connection.Subscribe(method, func(data []byte) []byte {
 		rs.logger.Trace().Msgf("Rpc server received request: %+v", data)
-
-		rpcRequest := serde.JsonRpcRequest[directfund.ObjectiveRequest]{}
+		rpcRequest := serde.JsonRpcRequest[T]{}
 		err := json.Unmarshal(data, &rpcRequest)
 		if err != nil {
 			panic("could not unmarshal direct fund objective request")
```

Comment:
I think we need a more generic message here, since we haven't drilled down to the type of objective request yet? Same goes for L76.

Reply:
[c69ad90](https://github.com/statechannels/go-nitro/pull/1117/commits/c69ad9013ec041768a0400d3b16953d670d7ad09)

Reply:
Good catch!